addon/services/router-scroll.js

```diff
@@ -84,10 +93,16 @@ const RouterScroll = Service.extend({
         set(this, 'targetElement', targetElement);
       }
 
-      const delayScrollTop = config.routerScroll.delayScrollTop;
+      const { whenPainted, whenIdle, delayScrollTop } = config.routerScroll;
       if (delayScrollTop === true) {
```

Comment:
If these properties default to `false` and only get set to `true` when set by the user, it might be simpler to use something like:
```
setProperties(this, {
  delayScrollTop,
  whenPainted,
  whenIdle
});
```